sensors/imu/mpu6000.c

```diff
@@ -189,73 +190,65 @@ static int mpu6000_hwSetup(mpu6000_ctx_t *ctx)
 }
 
 
-static void mpu6000_threadPublish(void *data)
+int mpu6000_read(const sensor_info_t *info, const sensor_event_t **evt)
 {
-	static uint32_t dAngleX = 0, dAngleY = 0, dAngleZ = 0;
-	static int32_t lastGyroX = 0, lastGyroY = 0, lastGyroZ = 0;
-	static time_t tStampLast;
+	mpu6000_ctx_t *ctx = info->ctx;
 
 	const uint8_t obuf = REG_DATA_OUT_ALL | SPI_READ_BIT;
-	sensor_info_t *info = (sensor_info_t *)data;
-	mpu6000_ctx_t *ctx = info->ctx;
 	uint8_t ibuf[SENSOR_OUTPUT_SIZE] = { 0 };
 	time_t tStamp;
-	float step;
 	int err;
 
-	gettime(&tStampLast, NULL);
+	/* data read */
+	err = sensorsspi_xfer(&ctx->spiCtx, &ctx->spiSS, &obuf, sizeof(obuf), ibuf, sizeof(ibuf), sizeof(obuf));
+	gettime(&tStamp, NULL);
+
+	if (err < 0) {
+		fprintf(stderr, "mpu6000 read: %d\n", err);
+		return -1;
+	}
+
+	/* Common package: gyro and accel utilize the same temperature reading */
+	ctx->evtAccel->accels.temp = translateTemp(ibuf[6], ibuf[7]);
+	ctx->evtGyro->gyro.temp = ctx->evtAccel->accels.temp;
+
+	ctx->evtAccel->accels.accelX = translateAcc(ibuf[0], ibuf[1]);
+	ctx->evtAccel->accels.accelY = translateAcc(ibuf[2], ibuf[3]);
+	ctx->evtAccel->accels.accelZ = translateAcc(ibuf[4], ibuf[5]);
+	ctx->evtAccel->timestamp = tStamp;
+
+	ctx->evtGyro->gyro.gyroX = translateGyr(ibuf[8], ibuf[9]);
+	ctx->evtGyro->gyro.gyroY = translateGyr(ibuf[10], ibuf[11]);
+	ctx->evtGyro->gyro.gyroZ = translateGyr(ibuf[12], ibuf[13]);
+	ctx->evtGyro->timestamp = tStamp;
+
+	*evt = &ctx->evt[0];
```

Comment:
Perhaps accept `evt == NULL` if use case is that it is optional

Reply:
It is mandatory. I can emphasize it in comment regarding the sensor API